pkg/network/common/dns.go

```diff
@@ -24,6 +27,9 @@ type dnsValue struct {
 	ttl time.Duration
 	// Holds (last dns lookup time + ttl), tells when to refresh IPs next time
 	nextQueryTime time.Time
+	// Used to know if DNS.Update or DNS.Add are modifying it, so that
+	// DNS.GetNextQueryTime can ignore it
+	updating bool
```

Comment:
should this be atomic?

Reply:
I don't think that's necessary because we're only modifying this variable when `DNS.lock` is held. So I think this is thread safe as is.

Reply:
> I don't think that's necessary because we're only modifying this variable when `DNS.lock` is held. So I think this is thread safe as is.

It's not necessary because you're only modifying _or reading_ the variable when the lock is held. It wouldn't be safe to read `updating` from `GetNextQueryTime` if you weren't holding the lock there.

Reply:
In fact, if the only reason we hold the lock is to update the "updating" bool, we should instead make it atomic and get rid of the locks.

Reply:
> In fact, if the only reason we hold the lock is to update the "updating" bool, we should instead make it atomic and get rid of the locks.

The problem is that this variable is part of a struct in a map and we cannot guarantee that the map entry won't be deleted while we update it without locking the whole map.

Reply:
I think that's fair. You cannot control the object's map membership by making the variable atomic.